src/main/java/uk/gov/justice/digital/job/filter/TimestampToDateFieldFilter.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.justice.digital.job.filter;
+
+import lombok.val;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+// Filter that converts Timestamp data in a field declared as a DateType into a date.
+public class TimestampToDateFieldFilter implements FieldFilter {
+
+    private static final Logger logger = LoggerFactory.getLogger(TimestampToDateFieldFilter.class);
+
+    private final Set<String> eligibleFields;
+
+    public TimestampToDateFieldFilter(StructType schema) {
+        // This filter applies to Timestamp fields.
+        eligibleFields = Arrays.stream(schema.fields())
+                .filter(f -> f.dataType() == DataTypes.DateType)
+                .map(StructField::name)
+                .collect(Collectors.toSet());
+    }
+
+    @Override
+    public boolean isEligible(String fieldName) {
+        return eligibleFields.contains(fieldName);
+    }
+
+    @Override
+    public Entry<String, Object> apply(Entry<String, Object> entry) {
+        val value = Optional.ofNullable(entry.getValue())
+                .map(Object::toString)
+                .orElse("");
+
+        // The raw date value may be represented as an ISO date time with a zeroed time part eg. 2012-01-01T00:00:00Z
+        // Split the string on T so we get the date before the T, and the time after the T.
+        // If we get two values we assume we're parsing a date and time, otherwise treat the value as a date.
+        val dateParts = Arrays.asList(value.split("T"));
+
+        if (dateParts.size() == 2) {
+            val dateString = dateParts.get(0);
+            val timeString = dateParts.get(1);
+            if (timeString != null && (timeString.replaceAll("[^1-9]", "").compareTo("0") > 0))
```

Comment:
Not quite, the regex leaves any digit over zero e.g. 

```scala> time
val res16: String = 01:01:01Z

scala> time.replaceAll("[^1-9]", "")
val res17: String = 111
```
So "111" is greater than "0" which tells us that time time value is set.

The aim here was to relatively cheaply figure out if the time is set or not. So if any non-zero values are present we'll trigger the warning log. (Originally this was a rejection). 

Obviously if the timestamp is legitimately set to midnight at 0s 000ms then we evaluate as false but this is acceptable since in practice we're processing a number of records so we should get a range of values where the time part is used.